projects/deltatheta/index.js

```diff
@@ -0,0 +1,86 @@
+const sdk = require('@defillama/sdk');
+const {
+  transformBscAddress,
+  transformPolygonAddress,
+} = require('../helper/portedTokens');
+const { getTokens } = require('../helper/getTokens');
+
+// Delta.theta Factory ABI (for needed calls)
+const factoryABI = require('./factory.abi');
+
+// Delta.theta Factory Address (On all chains)
+const FACTORY_ADDRESS = '0x000000000092126dc1bcec881165f92169733106';
+
+// TVL function generator (for BSC & POLYGON chains)
+function tvl(chain) {
+  const balances = {};
+
+  return async (_, __, chainBlocks) => {
+    // Prepare transform function for the selected chain
+    const transform = await (
+      chain === 'bsc'
+        ? transformBscAddress()
+        : transformPolygonAddress()
+    );
+
+    // Parse factory's pairs length
+    const pairsLength = (await sdk.api.abi.call({
```

Comment:
Hi, this looks good, but can you please chage it to use `calculateUniTvl` helper method instead. Reason being, lesser code duplication and it batch queries instead of single calls

Reply:
you can use this file as reference: https://github.com/DefiLlama/DefiLlama-Adapters/blob/69b8193cb6b4ed4715f2ce54d081ff28ee13e392/projects/radioshack/index.js

Reply:
Our contract is completely incompatible with Uniswap, so I just rewrote the SDK using multi call methods